flang/runtime/inquiry.cpp

```diff
@@ -76,5 +85,15 @@ std::int64_t RTDEF(SizeDim)(
   return static_cast<std::int64_t>(dimension.Extent());
 }
 
+void RTDEF(Shape)(void *result, const Descriptor &array, int kind) {
+  Terminator terminator{__FILE__, __LINE__};
```

Comment:
LGTM, but I would prefer passing relevant source information to the runtime.

Reply:
https://github.com/llvm/llvm-project/pull/94781